Review minimal function to match release testing instructions

sbesson opened the pull request:
See https://trello.com/c/nJ7SksJp/11-octave-test-script-packaging

This PR scripts the steps used for testing the Bio-Formats MATLAB/Octave package and reviews the top-level M file included to run these steps automatically.

With these changes, the release testing workflow should be limited to bumping the `VERSION` in the top-level `Dockerfile` and running  the following steps:

```
docker build -t bio-formats-octave .
docker run -ti bio-formats-octave minimal_test.m
```

The `test_bfopen.m` function was extracted from the upstream [bftest.m](https://github.com/openmicroscopy/bioformats/blob/v5.5.2/docs/sphinx/developers/examples/bftest.m) method used both for testing the MATLAB integration daily and the developers documentation. If this addition makes sense, a proposal would be to refactor this file upstream and bundle `test_bfopen.m` in the Octave package so that this can consume it directly and we only maintain examples in one place.

sbesson committed:
Expand minimal test to cover the release process

The goal of this PR is to expand the current minimal test shipped with the
Bio-Formats Octave container to be usable as part of the release process.

sbesson committed:
Update Travis command

sbesson committed:
Re-execute test in debugging mode

sbesson committed:
Use javaObject syntax

sbesson committed:
Use javaMethod

joshmoore commented on Dockerfile:
```diff
@@ -13,4 +13,7 @@ RUN echo "/home/octave/bioformats_package.jar" >> /home/octave/javaclasspath.txt
 
 RUN echo "pkg install bioformats-octave-$VERSION.tar.gz" | octave
+
+RUN wget https://downloads.openmicroscopy.org/images/OME-TIFF/2016-06/tubhiswt-3D.zip
```
Is it worth adding something like `?utm_source=docker`

sbesson replied:
Absolutely. Using `utm_source` or `--user-agent` as we do in `omego` would allow us to filter out these from the stats. Happy to go with `utm_source` if that's the best recommendation for now.

joshmoore replied:
We don't have anything in place for utm_source. I had forgotten the specifics of --user-agent so let's stick with that.

joshmoore commented:
We don't have anything in place for `utm_source`. I had forgotten the specifics of `--user-agent` so let's stick with that.

sbesson committed:
Add Docker user-agent when retrieving the images

sbesson committed:
Use latest image

sbesson committed:
Use legacy images via the deprecated-2017Q2 group

The recent deployment of Trusty images including a kernel update which affected
many applications invoking Java like Octave - see
https://bugs.launchpad.net/ubuntu/+source/linux/+bug/1699772 for more
information.

Until this bug is resolved by a new kernel update, this commit pins down the
version of the Ubuntu image used for testing the Bio-Formats Docker image in
Travis.

sbesson commented:
Note the failure in this PR is unrelated to the content but rather to the recent [Trusty images deployment](https://blog.travis-ci.com/2017-06-19-trusty-updates-2017-Q2). I can reproduce by re-running [a previous passing build](https://travis-ci.org/sbesson/bio-formats-octave-docker/builds/243575243) and can fix the status by [reverting to the legacy group](https://travis-ci.org/sbesson/bio-formats-octave-docker/builds/246373285) as suggested in the Travis blog post. I contacted Travis CI support for more information on how to fix this without using this group which will be deprecated in a few months.

sbesson commented:
Quick update. Thanks to @manics who pointed out the kernel update, it looks like this issue is related to the upstream https://bugs.launchpad.net/ubuntu/+source/linux/+bug/1699772 i.e. an issue with the latest kernel deployed in the Travis CI images with applications using the Java invokation API (like `pkg install bioformats` does).

sbesson committed:
Revert "Use latest image"

This reverts commit 752260732bc33d4137da22f7d219d1f5b146a89a.

dgault commented:
I tested this as part of the Bio-Formats 5.5.3 release testing and the script worked as expected. This is certainly a much more efficient method of enabling basic tests with Octave. The actual tests themselves seem sensible and provide suitable sanity tests.

This looks fine to me and Im happy to have this included as part of the integration tests.

sbesson commented:
Thanks @dgault. Merging and relaunching #10 with this included

joshmoore commented:
An aside: is the intent to have an image in the docker image for someone to test instantly? If not, a more conservative approach would be to download the image in .travis.yml (caching it) and then mount that directory into the test container passing an argument to `test_bfopen`.
